bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/PowerControlPower.java

```diff
@@ -29,31 +33,82 @@
+        final State state = handler.getSocket().getState();
         if (RefreshType.REFRESH == command) {
             handler.postUpdate(channelId, handler.getSocket().isConnected() ? OnOffType.ON : OnOffType.OFF);
-            return;
-        }
-        if (OnOffType.ON == command) {
-            logger.debug(
-                    "Received ON - Turning TV on via API is not supported by LG WebOS TVs. You may succeed using wake on lan binding, please consult lgwebos binding documentation.");
-        }
-        if (!handler.getSocket().isConnected()) {
-            /*
-             * Unable to send anything to a not connected device.
-             * onDeviceReady nor onDeviceRemoved will be called and item state would be permanently inconsistent.
-             * Therefore setting state to OFF
-             */
-            handler.postUpdate(channelId, OnOffType.OFF);
+        } else if (OnOffType.ON == command) {
+            if (State.DISCONNECTED == state) {
+                String macAddress = configProvider.getMacAddress();
+                if (macAddress.isEmpty()) {
+                    logger.debug("Received ON - Turning TV on via API is not supported by LG WebOS TVs. "
+                            + "You may succeed using wake on lan (WOL). "
+                            + "Please set the macAddress config value in Thing configuration to enable this.");
+                    handler.postUpdate(channelId, OnOffType.OFF);
+                } else {
+                    for (int i = 0; i < WOL_PACKET_RETRY_COUNT; i++) {
+                        scheduler.schedule(() -> {
+                            try {
+                                WakeOnLanUtility.sendWOLPacket(macAddress);
+                            } catch (IllegalArgumentException e) {
+                                logger.debug("Failed to send WOL packet: {}", e.getMessage());
+                            }
+                        }, i * WOL_PACKET_RETRY_DELAY_MILLIS, TimeUnit.MILLISECONDS);
+                    }
+                }
+            } else {
+                switch (state) {
+                    case CONNECTING:
+                    case REGISTERING:
+                    case REGISTERED:
+                        logger.debug("Received ON - TV is already on.");
+                        handler.postUpdate(channelId, OnOffType.ON);
+                        break;
+                    case DISCONNECTING:
+                        logger.debug("Received ON - TV is currently disconnecting.");
+                        handler.postUpdate(channelId, OnOffType.OFF);
+                        break;
+                    case DISCONNECTED:
+                        // none reachable code
+                        break;
+                }
+            }
         } else if (OnOffType.OFF == command) {
-            handler.getSocket().powerOff(getDefaultResponseListener());
+            if (State.REGISTERED == state || State.REGISTERING == state) {
```

Comment:
Please move this to the switch statement

Reply:
yes indeed